lms/static/scripts/frontend_apps/utils/test/api-test.js

```diff
@@ -651,3 +652,101 @@ describe('usePaginatedAPIFetch', () => {
     assert.equal(getMainContent(wrapper), 'No content');
   });
 });
+
+describe('usePolledAPIFetch', () => {
+  let fakeUseFetch;
+  let fakeRetry;
+  let fakeClearTimeout;
+
+  function mockFetchFinished() {
+    fakeUseFetch.returns({
+      data: {},
+      isLoading: false,
+      retry: fakeRetry,
+    });
+  }
+
+  function mockLoadingState() {
+    fakeUseFetch.returns({
+      data: null,
+      isLoading: true,
+    });
+  }
+
+  beforeEach(() => {
+    fakeUseFetch = sinon.stub();
+    mockLoadingState();
+
+    fakeRetry = sinon.stub();
+    fakeClearTimeout = sinon.stub();
+
+    const fakeUseConfig = sinon.stub();
+    fakeUseConfig.returns({
+      api: { authToken: 'some-token' },
+    });
+
+    $imports.$mock({
+      '../config': { useConfig: fakeUseConfig },
+      './fetch': { useFetch: fakeUseFetch },
+    });
+  });
+
+  afterEach(() => {
+    $imports.$restore();
+  });
+
+  function TestWidget({ shouldRefresh }) {
+    const result = usePolledAPIFetch({
+      path: '/api/some/path',
+      shouldRefresh,
+
+      // Keep asynchronous nature of mocked setTimeout, but with a virtually
+      // immediate execution of the callback
+      _setTimeout: callback => setTimeout(callback, 1),
```

Comment:
You might be able to use `setTimeout(callback, 0)` here. This will be fractionally faster and still be async.